tests/wpunit/TEC/Events/Admin/Notice/RESTAPITest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * REST API Notice test.
+ */
+
+declare( strict_types=1 );
+
+namespace TEC\Tests\Events\Admin\Notice;
+
+use Codeception\TestCase\WPTestCase;
+use TEC\Events\Admin\Notice\Rest_Api;
+use Tribe\Tests\Traits\With_Uopz;
+use WP_Error;
+
+/**
+ * Class RESTAPITest
+ *
+ * @since TBD
+ */
+class RESTAPITest extends WPTestCase {
+
+	use With_Uopz;
+
+	/**
+	 * Clean up after this test class finishes.
+	 *
+	 * @return void
+	 */
+	public function wpTearDownAfterClass() {
+		remove_all_filters( 'tec_events_site_is_development_mode' );
+		remove_all_filters( 'tec_events_rest_api_response_blocked_due_to_timeout' );
+	}
+
+	/**
+	 * Test scenarios that should report whether the API is blocked.
+	 *
+	 * @test
+	 * @return void
+	 */
+	public function should_correctly_determine_if_the_api_is_blocked() {
+		$request_return_success = function() {
+			return [ 'response' => [ 'code' => 200 ] ];
+		};
+
+		$request_return_timeout = function() {
+			return new WP_Error(
+				'http_request_failed',
+				'cURL error 28: Operation timed out after 10000 milliseconds with 0 bytes received'
+			);
+		};
+
+		$request_return_ssl_error = function() {
+			return new WP_Error(
+				'http_request_failed',
+				'cURL error 60: SSL certificate problem: unable to get local issuer certificate'
+			);
+		};
+
+		$rest_api = new Rest_Api();
+
+		// A timeout should not be blocking, unless we filter it to true.
+		$this->set_fn_return( 'wp_safe_remote_get', $request_return_success, true );
+		$this->assertFalse( $rest_api->is_rest_api_blocked( true ) );
+		$this->unset_uopz_returns();
```

Comment:
Shouldn't be here - its being called automatically after every test. If you check the trait's code you will see an `@after` annotation. That means it runs after every test. Similar there are `@before`, `@beforeClass` and `@afterClass`

Reply:
Ah, thanks! I missed that.

What about when I'm overriding the same function multiple times?

Reply:
If you override - you should call the parent every time. Better to just create your own unique methods and annotate them with `@before` or `@after`

Reply:
@dpanta94 I mean the use of `unset_uopz_returns()`. I'm overriding `wp_safe_remote_get()` multiple times in succession. It looks like it just adds to an unindexed array each time.

Reply:
Ah sorry @JPry, yes your latest overwrite will be the one used